src/main/java/org/openpnp/machine/reference/ReferenceNozzleTipCalibration.java

```diff
@@ -1017,8 +1028,13 @@ public void setEnabled(boolean enabled) {
         this.enabled = enabled;
     }
 
-    public CvPipeline getPipeline() throws Exception {
-        pipeline.setProperty("camera", VisionUtils.getBottomVisionCamera());
+    public CvPipeline getPipeline(Camera camera, Location measureLocation) throws Exception {
+        pipeline.setProperty("camera", camera);
+        pipeline.setProperty("nozzleTip.diameter", getCalibrationTipDiameter());
+        pipeline.setProperty("nozzleTip.tolerance", getOffsetThresholdLength().multiply(1.4));
```

Comment:
Why is this multiplied by 1.4?

Reply:
>     * Nozzle Tip Diameter: Can we change the name and be explicit? If this is specifically a vision property, call it Vision Diameter, or something like that. If it is specifically the outer diameter that we want, call it Outer Diameter.

I'll call it Vision Diameter. It is not necessarily the outer diameter, i.e. from the Instruction for Use:

> **Vision Diameter**: Must be set to the physical diameter of the feature you want to detect. Typically, a feature at the very point of the nozzle tip. Either the outer diameter or the air bore. ...

_Mark

Reply:
> Why is this multiplied by 1.4?

I must give it some margin to robustly detect the nozzle tip even outside the threshold, then causing an exception. Otherwise, it would possible "detect" a random speck of the background as the nozzle tip, falsely assuming it is inside the threshold. 

This typically happens in camera position/rotation calibration with the nozzle tip being off-center, if your Units per Pixel are off e.g. because the focal plane for the bottom camera Z has changed or when using a different video mode etc. Better to get "positive falses" than "false positives" (hope this translates to English) ;-) 

_Mark

Reply:
I should have been more direct: Can you add a comment in the code explaining this magic number? :)

Reply:
You're right. Made it configurable through `machine.xml` if absolutely needed and gave it some comments.

Reply:
Are these changes ok?

Reply:
The mailer does not represent this well. What I meant are 29183eb and 2a28ff8 (hope this will be hyperlinked as usual).